### ISSUE TITLE
[Bug]vendored framework not linking if name contains -

### ISSUE THREAD

**Issue by businessengine** [CocoaPods/CocoaPods#8584]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

When a vendored framework whose name contains -, CocoaPods will fail to include that framework in
```
        spec.ios.vendored_frameworks = ["VendoredFrameworks/a-frame-work.framework"]
```

## What did you do?

Run `pod install`

## What did you expect to happen?
OTHER_LDFLAGS should contains`framework "a-frame-work"`

## What happened instead?
`framework "a-frame-work"` is missing

### Stack

```
   CocoaPods : 1.6.1
        Ruby : ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin16]
    RubyGems : 2.6.11
        Host : Mac OS X 10.14.3 (18D109)
       Xcode : 10.1 (10B61)
         Git : git version 2.17.2 (Apple Git-113)
Ruby lib dir : /usr/local/Cellar/ruby/2.4.1_1/lib
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.3
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
```

**Comment by businessengine** [comment]
Seems related to this one https://github.com/CocoaPods/CocoaPods/issues/8519

**Comment by dnkoutso** [comment]
@businessengine close as dup?

**Comment by businessengine** [comment]
Yep, closed as duplicate #8519
